packages/flutter/lib/src/foundation/basic_types.dart

```diff
@@ -238,3 +238,10 @@ class Factory<T> {
     return 'Factory(type: $type)';
   }
 }
+
+/// Linearly interpolate between two `Duration`s.
+Duration lerpDuration(Duration a, Duration b, double t) {
+  return Duration(
+    microseconds: (a.inMicroseconds + (b.inMicroseconds - a.inMicroseconds) * t).round(),
+  );
+}
```

Comment:
Here and below: Very minor nit, source files should [end with a linefeed](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline). That said, that's probably more pedantic than anything -- I'm 99% certain that the entire Dart toolchain consistently handles this just fine without it. But not a bad idea anyway.